java-sdk/src/main/java/com/netflix/conductor/sdk/executor/WorkflowExecutor.java

```diff
@@ -0,0 +1,370 @@
+
+        if(serverProcess != null) {
+            return;
+        }
+
+        String configFile = WorkflowExecutor.class.getResource("/test-server.properties").getFile();
+        String tempDir = System.getProperty("java.io.tmpdir");
+        Path serverFile = Paths.get(tempDir, "conductor-server.jar");
+        if(!Files.exists(serverFile)) {
+            Files.copy(new URL(repositoryURL).openStream(), serverFile);
+        }
+        serverProcess = Runtime.getRuntime().exec("java -Dserver.port=" + localServerPort + " -DCONDUCTOR_CONFIG_FILE=" + configFile + " -jar " + serverFile.toString());
+        InputStream in = serverProcess.getInputStream();
+        InputStream err = serverProcess.getErrorStream();
+        BufferedReader error = new BufferedReader(new InputStreamReader(serverProcess.getErrorStream()));
+        BufferedReader op = new BufferedReader(new InputStreamReader(serverProcess.getInputStream()));
+
+        Executors.newSingleThreadScheduledExecutor().execute(()->{
+
+            String line = null;
+            while(true) {
+                try {
+                    if (!((line=error.readLine()) != null)) break;
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                //copy to standard error
+                System.err.println(line);
+            }
+        });
+
+        Executors.newSingleThreadScheduledExecutor().execute(()->{
+
+            String line = null;
+            while(true) {
+                try {
+                    if (!((line=op.readLine()) != null)) break;
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                //copy to standard out
+                System.out.println(line);
+            }
+        });
+
+    }
+
+    public void shutdown() {
+        if(serverProcess != null) {
+            serverProcess.destroyForcibly();
+        }
+        if(taskRunner != null) {
+            taskRunner.shutdown();
+        }
+        ses.shutdown();
+    }
+
+    /**
+     * Starts a workflow execution and waits till
+     * @param name
+     * @param version
+     * @param input
+     * @return
+     */
+    public Workflow executeWorkflow(String name, int version, Map<String, Object> input) {
+        StartWorkflowRequest request = new StartWorkflowRequest();
+        request.setInput(input);
+        request.setName(name);
+        request.setVersion(version);
+
+        String workflowId = workflowClient.startWorkflow(request);
+        CountDownLatch latch = new CountDownLatch(1);
+        runningWorkflows.put(workflowId, latch);
+        Uninterruptibles.awaitUninterruptibly(latch);
+        runningWorkflows.remove(workflowId);
+        return workflowClient.getWorkflow(workflowId, true);
+    }
+
+
+
+    TypeReference<List<TaskDef>> listOfTaskDefs = new TypeReference<List<TaskDef>>() {};
```

Comment:
Consider moving this up to the same location as other class variables?

Reply:
Made this fix